query/src/main/java/org/infinispan/query/indexmanager/AbstractUpdateCommand.java

```diff
@@ -70,14 +71,15 @@ public boolean isReturnValueExpected() {
     */
    @Override
    public void fetchExecutionContext(CommandInitializer ci) {
-      if (ci.getCacheManager().cacheExists(cacheName)) {
-         Cache cache = ci.getCacheManager().getCache(cacheName);
+      String name = cacheName.toString();
```

Comment:
I guess I spoke to soon about `ByteString.toString()` not being used... I guess here we need a way for the per-cache inbound invocation handler to pass the `Cache` to the command.